test/unit/abilities/provider_any_test.rb

```diff
@@ -71,57 +81,113 @@ def test_policies_not_tenant
-        user.member_permission_ids = [:monitoring]
-        user.member_permission_service_ids = []
+    test "member has :partners and access to a service can't show AccountRelatedEvent if there is another service" do
+      service = FactoryBot.create(:simple_service, account: @account)
+      user.member_permission_service_ids = [service.id]
+      user.member_permission_ids = [:partners]
 
-        assert_cannot ability, :show, limit_violation_reached_provider_event
-      end
+      assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(@buyer, @buyer_user)
+    end
+
+    test "member has :partners and access to no service can't show AccountRelatedEvent if there is a service" do
+      user.member_permission_service_ids = []
+      user.member_permission_ids = [:partners]
 
-      test 'can show AlertRelatedEvent when user has :monitoring and has access to the service' do
-        user.member_permission_ids = [:monitoring]
-        user.member_permission_service_ids = [limit_violation_reached_provider_event.service.id]
+      assert_cannot ability, :show, Accounts::AccountCreatedEvent.create(@buyer, @buyer_user)
+    end
 
-        assert_can ability, :show, limit_violation_reached_provider_event
-      end
+    test 'member has :partners and access to one service can show AccountRelatedEvent if there is that service among others' do
+      service = FactoryBot.create(:simple_service, account: @account)
+      service_plan = FactoryBot.create(:simple_service_plan, issuer: service)
+      FactoryBot.create(:service_contract, plan: service_plan, user_account: @buyer)
+
+      user.member_permission_service_ids = [service.id]
+      user.member_permission_ids = [:partners]
+
+      assert_can ability, :show, Accounts::AccountCreatedEvent.create(@buyer, @buyer_user)
     end
+  end
 
-    private
+  class ShowAlertRelatedEventTest < BaseTest
+    setup do
+      service = FactoryBot.create(:simple_service, account: @account)
+      plan = FactoryBot.create(:simple_application_plan, issuer: service)
+      cinstance = FactoryBot.create(:cinstance, plan: plan)
+      alert = FactoryBot.create(:limit_violation, cinstance: cinstance)
+      @limit_violation_reached_provider_event = Alerts::LimitViolationReachedProviderEvent.create(alert)
+    end
 
-    def ability
-      Ability.new(user)
+    attr_reader :limit_violation_reached_provider_event
+
+    test 'cannot show AlertRelatedEvent when user does not have :monitoring' do
+      assert_not user.has_permission? :monitoring
+      assert_cannot ability, :show, limit_violation_reached_provider_event
+    end
+
+    # If the user has service-related permission, and no :services permission, it means it has access to all services
+    test 'can show AlertRelatedEvent when user has :monitoring and the user has access to all services' do
+      user.member_permission_ids = [:monitoring]
+      assert user.has_permission? :monitoring
```

Comment:
Isn't this assertion redundant?
```suggestion
      user.member_permission_ids = [:monitoring]
```

Reply:
I removed it: https://github.com/3scale/porta/pull/4029/commits/3475447f6a596296c036ec21dee126b6a51a78ff